### PR TITLE
Added inPolygon with tests

### PR DESCRIPTION
**Pull request by ArneSpang**
Added inpolygon function which is needed for the setup generation of Tatjana/Andrea

### CODE REVIEW

File: src/utils.jl [hunk 1]
```diff
@@ -1640,4 +1641,106 @@ function LithostaticPressure!(Plithos::Array{T,N}, Density::Array{T,N}, dz::Numb
...
     return nothing
+end
+
+"""
+    inPolygon!(PolyX::Vector, PolyY::Vector, X::Matrix, Y::Matrix, INSIDE::Matrix; fast=false)
+
+Checks if points given by matrices `X` and `Y` are in or on (both cases return true) a polygon given by `PolyX` and `PolyY`. Boolean `fast` will trigger faster version that may miss points that are exactly on the edge of the polygon. Speedup is a factor of 3.
+
+"""
+function inPolygon!(PolyX::Vector{T}, PolyY::Vector{T}, X::Matrix{T}, Y::Matrix{T}, INSIDE::Matrix{Bool}; fast=false) where T <: Real
```
Comment (boriskaus) [inline src/utils.jl:1652]: the naming convention for in-place Julia functions is that the array that is being changed should be listed first.
So this should be renamed as:
```Julia
function inPolygon!(INSIDE::Matrix{Bool}, PolyX::Vector{T}, PolyY::Vector{T}, X::Matrix{T}, Y::Matrix{T}; fast=false) where T <: Real
```

**Review by boriskaus — APPROVED** [review]

**Review by boriskaus — CHANGES_REQUESTED** [review]
the input to the `inPolygon!` function should be changed

**Review by boriskaus — APPROVED** [review]

File: src/utils.jl (hunk 1 above)
Comment (ArneSpang) [reply src/utils.jl]: I changed it around but now one of the tests failed. It says it is in IO and sounds like a network problem of the test server. Do you understand what is going on there?
image

Comment (boriskaus) [reply src/utils.jl]: yes, we have this frequently. Downloading remote data is sometimes fragile. restarting the CI often solves it.

**Pull request merged**